tests/core/middleware/test_http_request_retry.py

```diff
@@ -0,0 +1,60 @@
+import pytest
+from unittest.mock import Mock
+
+from web3.middleware import http_retry_request_middleware
+from web3.middleware.http_retry_request import check_method
+from requests.exceptions import ConnectionError
+from web3.providers import HTTPProvider, IPCProvider
+
+
+@pytest.fixture
+def http_retry_request_setup():
+    make_request, web3 = Mock(), Mock()
+    setup = http_retry_request_middleware(make_request, web3)
+    setup.web3 = web3
+    setup.make_request = make_request
+    setup.check_method = check_method
+    return setup
```

Comment:
This mock testing doesn't cover all our needs (and might be too brittle as the internal implementations change over time).

If you must mock, maybe mock out something much deeper, like `make_post_request`, so that all the HTTPProvider and its new built-in middlewares get tested.

Reply:
Changed this and added some tests to make sure the whole process works.